### PR TITLE
http2: general code cleanups

### PR DESCRIPTION
**Pull request by jasnell**
A number of code cleanups that I've been accumulating

##### Checklist
- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-guidelines)

##### Affected core subsystem(s)
http2

### CODE REVIEW

File: lib/http2.js [hunk 1]
```diff
@@ -6,17 +6,20 @@ process.emitWarning(
...
 module.exports = {
-  constants: core.constants,
-  getDefaultSettings: core.getDefaultSettings,
-  getPackedSettings: core.getPackedSettings,
-  createServer: core.createServer,
-  createSecureServer: core.createSecureServer,
-  createServerSession: core.createServerSession,
-  connect: core.connect,
-  secureConnect: core.secureConnect
+  constants,
```
Comment (sebdeckers) [inline lib/http2.js:19]: Why are `constants`, `getDefaultSettings`, and `getPackedSettings` exported?

File: lib/internal/http2/core.js [hunk 2]
```diff
@@ -443,64 +325,101 @@ function setupHandle(session, socket, type, options, settings) {
...
 
+// Upon creation, the Http2Session takes ownership of the socket. The session
+// may not be ready to use immediately if the socket is not yet fully connected.
 class Http2Session extends EventEmitter {
-  constructor(type, options, socket, settings) {
+
+  // type     { number } either NGHTTP2_SESSION_SERVER or NGHTTP2_SESSION_CLIENT
+  // options  { Object }
+  // socket   { net.Socket | tls.TLSSocket }
+  // settings { Object }
```
Comment (sebdeckers) [inline lib/internal/http2/core.js]: Remove settings line from comment

**Review by sebdeckers — COMMENTED** [review]
Great stuff!

File: lib/http2.js (hunk 1 above)
Comment (jasnell) [reply lib/http2.js:19]: They are all public APIs. `constants` includes constants that will be used by end users, `getDefaultSettings()` returns a pre-filled object that can be used by users to start with the default settings then modified, `getPackedSettings()` returns a buffer suitable for use with the `HTTP2-Settings` header field. These are specifically intended for end users to make use of.

Comment (sebdeckers) [reply lib/http2.js:19]: Thanks for clarifying. Haven't played with the settings much yet, still just doing basic stuff. 🤓

My concern with exposing the constants is that they contain many like `NGHTTP2_*`. Seems like something internal to the implementation, to be shielded from consumers.

**Review by mcollina — CHANGES_REQUESTED** [review]
@jasnell have you checked if this effect throughput? I'm a bit concerned by all those `defineProperty` calls.

**Comment by jasnell** [comment]
@mcollina ... yes, benchmark shows no significant change. In fact, setting the properties this way provides a slight perf improvement over getting at the data through getters.

**Review by mcollina — APPROVED** [review]
LGTM

**Comment by jasnell** [comment]
Landed!

**Pull request closed without merging**